static/js/markdown.js

```diff
@@ -149,6 +149,36 @@ function handleAvatar(email) {
            ' title="' + email + '">';
 }
 
+function handleTimestamp(time) {
+    var timeobject;
+    if (isNaN(time)) {
+        timeobject = moment(time); // not a Unix timestamp
+    } else {
+        // JavaScript dates are in milliseconds, Unix timestamps are in seconds
+        timeobject = moment(time * 1000);
+    }
+    var timestring = "";
+    var istimevalid = !(timeobject === null || !timeobject.isValid());
+
+    // outer span tag
+    timestring += '<span class="timestamp"';
+    if (istimevalid) {timestring += ' value="' + timeobject.unix() + '"';}
+    timestring += '>';
+    // emoji string
+    timestring += '<img alt=":clock4:" class="emoji" ';
+    timestring += 'src="/static/generated/emoji/images/emoji/unicode/1f553.png" ';
+    timestring += 'title=":clock4:">';
+    // inner span tag
+    timestring += '<span';
+    if (istimevalid) {timestring += ' value="' + timeobject.unix() + '"';}
+    timestring += '>';
+    // actual text by user
+    timestring += time;
+    // closing tags
+    timestring += '</span></span>';
+    return timestring;
```

Comment:
This code feels really messy, but I guess my proposed tweaks to the syntax will eliminate most of it?